libpod/runtime_pod.go

```diff
@@ -65,6 +78,36 @@ func (r *Runtime) LookupPod(idOrName string) (*Pod, error) {
 	return r.state.LookupPod(idOrName)
 }
 
+// GetPauseContainer returns a the pause container for a pod.
+// If the container returned is nil, the pod has no pause container.
+func (r *Runtime) GetPauseContainer(p *Pod) (*Container, error) {
```

Comment:
This should be a pod method - func (p *Pod) GetPauseContainer(). Store the ID of the pause container in pod state, and retrieve just the one container based on that ID.